sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -1966,6 +1966,49 @@ pi_result cuda_piDeviceGetInfo(pi_device device, pi_device_info param_name,
                         name.data());
   }
 
+  case PI_DEVICE_INFO_MAX_MEM_BANDWIDTH: {
+    int memory_clock_khz = 0;
+    sycl::detail::pi::assertion(
+        cuDeviceGetAttribute(&memory_clock_khz, CU_DEVICE_ATTRIBUTE_MEMORY_CLOCK_RATE,
+                             device->get()) == CUDA_SUCCESS);
+
+    int memory_bus_width = 0;
+    sycl::detail::pi::assertion(
+        cuDeviceGetAttribute(&memory_bus_width,
+                             CU_DEVICE_ATTRIBUTE_GLOBAL_MEMORY_BUS_WIDTH,
+                             device->get()) == CUDA_SUCCESS);
+
+    // Reference
+    // https://github.com/jeffhammond/HPCInfo/blob/master/cuda/gpu-detect.cu
+    int major = 0;
+    sycl::detail::pi::assertion(
+        cuDeviceGetAttribute(&major,
+                             CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MAJOR,
+                             device->get()) == CUDA_SUCCESS);
+
+    int minor = 0;
+    sycl::detail::pi::assertion(
+        cuDeviceGetAttribute(&minor,
+                             CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MINOR,
+                             device->get()) == CUDA_SUCCESS);
+
+    // Xavier AGX override
+    if (major == 7 && minor == 2) {
+       memory_clock_khz = 2133000;
+    }
+
+    // Orin AGX override
+    if (major == 8 && minor == 7) {
+       memory_clock_khz = 3200000;
+       memory_bus_width = 256;
+    }
```

Comment:
Since this is overwriting the values read from the previous info calls, could we just not do the calls on these specific devices? I'm thinking something like:

```suggestion
    int major = 0;
    sycl::detail::pi::assertion(
        cuDeviceGetAttribute(&major,
                             CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MAJOR,
                             device->get()) == CUDA_SUCCESS);

    int minor = 0;
    sycl::detail::pi::assertion(
        cuDeviceGetAttribute(&minor,
                             CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MINOR,
                             device->get()) == CUDA_SUCCESS);

    // Some specific devices seem to need special handling. See reference
    // https://github.com/jeffhammond/HPCInfo/blob/master/cuda/gpu-detect.cu
    bool is_xavier_agx = major == 7 && minor == 2;
    bool is_orin_agx = major == 7 && minor == 2;

    int memory_clock_khz = 0;
    if (is_xavier_agx) {
        memory_clock_khz = 2133000;
    } else if (is_orin_agx) {
        memory_clock_khz = 3200000;
    } else {
        sycl::detail::pi::assertion(
            cuDeviceGetAttribute(&memory_clock_khz, CU_DEVICE_ATTRIBUTE_MEMORY_CLOCK_RATE,
                                device->get()) == CUDA_SUCCESS);
    }
    
    int memory_bus_width = 0;
    if (is_orin_agx) {
       memory_bus_width = 256;
    } else {
        sycl::detail::pi::assertion(
            cuDeviceGetAttribute(&memory_bus_width,
                                CU_DEVICE_ATTRIBUTE_GLOBAL_MEMORY_BUS_WIDTH,
                                device->get()) == CUDA_SUCCESS);
    }
```

Reply:
I don't know how to slightly update your change before 'commit suggestion'. Thank you. 

 bool is_orin_agx = major == 7 && minor == 2;  =>  bool is_orin_agx = major == 8 && minor == 7;

Reply:
I can edit it, but I fear the formatting is off anyway so feel free to copy-paste it and commit it yourself.